share/config.py

```diff
@@ -394,40 +391,57 @@ def include_exclude_filter(self, value: IncludeExcludeFilter) -> None:
 
         self._include_exclude_filter = value
 
-    def get_output_by_type(self, output_type: str) -> Optional[Output]:
+    def get_output_by_destination(self, output_destination: str) -> Optional[Output]:
         """
         Output getter.
-        Returns a specific output given its type
+        Returns a specific output given its destination
         """
 
-        return self._outputs[output_type] if output_type in self._outputs else None
+        return self._outputs[output_destination] if output_destination in self._outputs else None
 
-    def get_output_types(self) -> list[str]:
+    def get_output_destinations(self) -> list[str]:
         """
-        Output types getter.
-        Returns all the defined output types
+        Output destinations getter.
+        Returns all the defined output destinations
         """
 
         return list(self._outputs.keys())
 
-    def delete_output_by_type(self, output_type: str) -> None:
+    def delete_output_by_destination(self, output_destination: str) -> None:
         """
         Output deleter.
         Delete a defined output by its type
         """
 
-        del self._outputs[output_type]
+        del self._outputs[output_destination]
 
     def add_output(self, output_type: str, **kwargs: Any) -> None:
         """
         Output setter.
         Set an output given its type and init kwargs
         """
-        if not isinstance(output_type, str):
-            raise ValueError("`type` must be provided as string")
 
-        if output_type in self._outputs:
-            raise ValueError(f"Duplicated `type` {output_type}")
+        if output_type not in _available_output_types:
```

Comment:
Same here to chaneg type with destination

Reply:
Same comment as before, `output.type` can only be elasticsearch or logstash, we use it to create the shipper, not the same as the destination